Sources/PandaSDK/AppStoreService.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  AppStoreService.swift
+//  
+//
+//  Created by Denys Danyliuk on 29.08.2023.
+//
+
+import StoreKit
+
+final class AppStoreService: NSObject {
+    private var products: [String: Product] = [:]
+    private var task: Task<Void, Error>?
+    private var testTask: Task<Void, Error>?
+    private let verificationClient: VerificationClient
+    
+    var onVerify: ((_ source: PaymentSource?) async -> Void)?
+    var purchasing: Bool = false
+    
+    init(verificationClient: VerificationClient) {
+        self.verificationClient = verificationClient
+         
+        super.init()
+    }
+    
+    deinit {
+        task?.cancel()
+    }
+    
+    func startTask() {
+        task = Task {
+            for await transactions in debounceCollect(Transaction.updates, for: 2) {
+                guard !purchasing else {
+                    return
+                }
+                await onVerify?(nil)
+                for verifiedTransaction in transactions {
+                    guard let transaction = try? verifiedTransaction.payloadValue else {
+                        return
+                    }
+                    await transaction.finish()
+                }
+            }
+        }
+        
+        // Subscribe to shouldAddStorePayment
+        SKPaymentQueue.default().add(self)
+    }
+    
+    func restore() async throws {
+        try await AppStore.sync()
+    }
+    
+    func fetchProducts(productIDs: Set<String>, completion: @escaping (Result<[Product], Error>) -> Void) {
+        Task { [weak self] in
+            do {
+                let storeProducts = try await Product.products(for: productIDs)
+                for product in storeProducts {
+                    self?.products[product.id] = product
+                }
+                completion(.success(storeProducts))
+            } catch {
+                pandaLog("FetchProduct Error: \(error)")
+                completion(.failure(error))
+            }
+        }
+    }
+    
+    func receiptBase64String() throws -> String {
+        guard let appStoreReceiptURL = Bundle.main.appStoreReceiptURL else {
+            throw Errors.message("Missing appStoreReceiptURL")
+        }
+        do {
+            return try Data(contentsOf: appStoreReceiptURL).base64EncodedString()
+        } catch {
+            throw Errors.appStoreReceiptError(error)
+        }
+    }
+
+    func purchase(productID: String, source: PaymentSource) async throws -> PurchaseResult {
+        purchasing = true
+        let product = try await getProduct(with: productID)
+        let result = try await product.purchase(options: [
+            .simulatesAskToBuyInSandbox(false)
+        ])
+        
```

Comment:
If you got a error, as result you will see infinity loading? Because you set `purchasing` to true, but i dont see code which set `purchasing` to false